camply/search/base_search.py

```diff
@@ -140,9 +140,12 @@ def _get_intersection_date_overlap(self, date: Timestamp, periods: int) -> bool:
         -------
         bool
         """
-        timestamp_range: List[Timestamp] = date_range(
-            start=date.to_pydatetime(), periods=periods
-        )
+        timestamp_range = []
+        if type(date) is Timestamp:
+            timestamp_range = date_range(start=date.to_pydatetime(), periods=periods)
+        else:
+            timestamp_range = date_range(start=date, periods=periods)
```

Comment:
Nice - I wish I could easily remember exactly where my datetime objects get converted into Timestamps and then become difficult to deal with.

Reply:
I originally I remember intending to track that down and factoring this code out. I vaguely recall this having something to do with dates coming from a config file? Does that ring any bells?

Reply:
I think there are a couple places where I convert an array of `AvailableCampsite` objects into a pandas DataFrame, do some manipulation, and then transform them back into an array of `AvailableCampsite` - somehow tunring them into [Timestamp](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Timestamp.html#pandas-timestamp) along the way. This might be the big offender here: https://github.com/juftin/camply/blob/173ee4792dbee1fb922438477697780631986932/camply/search/base_search.py#L752-L769